app/lib/cashu/types.ts

```diff
@@ -0,0 +1,127 @@
+ * of spending condition, proofs may be treated as a regular anyone-can-spend tokens.
+ * Applications need to make sure to check whether the mint supports a specific kind of
+ * spending condition by checking the mint's info endpoint.
+ */
+export const WELL_KNOWN_SECRET_KINDS = ['P2PK'] as const;
+
+/**
+ * the kind of the spending condition
+ */
+export type WellKnownSecretKind = (typeof WELL_KNOWN_SECRET_KINDS)[number];
+
+/**
+ * The data from a parsed stringified NUT-10 secret
+ */
+export type NUT10SecretData = {
+  nonce: string;
+  data: string;
+  tags?: NUT10SecretTag[];
+};
+
+/**
+ * The raw NUT-10 secret from parsing the proof secret that describes the spending conditions
+ * of the proof.
+ * @example
+ * ```json
+ * ["P2PK", {
+ *   "nonce": "859d4935c4907062a6297cf4e663e2835d90d97ecdd510745d32f6816323a41f",
+ *   "data": "0249098aa8b9d2fbec49ff8598feb17b592b986e62319a4fa488a3dc36387157a7",
+ *   "tags": [["sigflag", "SIG_INPUTS"]]
+ * }]
+ * ```
+ */
+export type ParsedNUT10Secret = [WellKnownSecretKind, NUT10SecretData];
+
+/**
+ * A NUT-10 secret in a proof is stored as a JSON string of a tuple:
+ * [kind, {nonce, data, tags?}]
+ *
+ * When parsed, it is transformed into this object format.
+ * @example
+ * ```json
+ * {
+ *   "secret": "[\"P2PK\", {
+ *     \"nonce\": \"859d4935c4907062a6297cf4e663e2835d90d97ecdd510745d32f6816323a41f\",
+ *     \"data\": \"0249098aa8b9d2fbec49ff8598feb17b592b986e62319a4fa488a3dc36387157a7\",
+ *     \"tags\": [[\"sigflag\", \"SIG_INPUTS\"]]
+ *   }]"
+ * }
+ * ```
+ *
+ * Gets parsed into:
+ * ```json
+ * {
+ *   "kind": "P2PK",
+ *   "data": "0249098aa8b9d2fbec49ff8598feb17b592b986e62319a4fa488a3dc36387157a7",
+ *   "nonce": "859d4935c4907062a6297cf4e663e2835d90d97ecdd510745d32f6816323a41f",
+ *   "tags": [["sigflag", "SIG_INPUTS"]]
+ * }
+ * ```
+ */
+export type NUT10Secret = {
+  /**
+   * well-known secret kind
+   * @example "P2PK"
+   */
+  kind: WellKnownSecretKind;
+  /**
+   * Expresses the spending condition specific to each kind
+   * @example "0249098aa8b9d2fbec49ff8598feb17b592b986e62319a4fa488a3dc36387157a7"
+   */
+  data: string;
+  /**
+   * A unique random string
+   * @example "859d4935c4907062a6297cf4e663e2835d90d97ecdd510745d32f6816323a41f"
+   */
+  nonce: string;
+  /**
+   * Hold additional data committed to and can be used for feature extensions
+   * @example [["sigflag", "SIG_INPUTS"]]
+   */
+  tags?: NUT10SecretTag[];
+};
+
+/**
+ * A plain secret is a random string
+ *
+ * @see https://github.com/cashubtc/nuts/blob/main/00.md for plain string secret format
+ */
+export type PlainSecret = string;
+
+/**
+ * A proof secret can be either be a random string or a NUT-10 secret
+ *
+ * @see https://github.com/cashubtc/nuts/blob/main/10.md for NUT-10 secret format
+ * @see https://github.com/cashubtc/nuts/blob/main/00.md for plain string secret format
+ */
+export type ProofSecret = NUT10Secret | PlainSecret;
```

Comment:
Do you think that we should call `NUT10Secret`s `WellKnownSecret`. I like NUT10Secret because it's more clear if you're familiar with the NUTs, but for someone that isn't familiar with the spec then `WellKnownSecret` is probably more meaningful.

 I called the secret described in NUT-00 `PlainSecret`, but maybe that should be `NUT00Secret`.  NUT 00 is a lot more than just describing what a secret is, so `NUT00Secret` isn't as meaningful for me

Reply:
current names seem fine to me